torch/CMakeLists.txt

```diff
@@ -340,13 +340,19 @@ endif()
 
 
 add_library(torch_python SHARED ${TORCH_PYTHON_SRCS})
+if(DEFINED ENV{BUILD_CAFFE2})
+  string(TOUPPER "$ENV{BUILD_CAFFE2}" BUILD_CAFFE2_STR)
+  if("${BUILD_CAFFE2_STR}" STREQUAL "1" OR "${BUILD_CAFFE2_STR}" STREQUAL "ON" OR "${BUILD_CAFFE2_STR}" STREQUAL "TRUE")
+    string(APPEND CMAKE_CXX_FLAGS " -DBUILD_CAFFE2")
```

Comment:
`string(APPEND CMAKE_CXX_FLAGS` should not be used anywhere other than at top level file, as it indiscriminately applies this flag to all targets, including third party libraries. 
Please use something more targeted, like  `target_compile_definitions` or `set_source_files_properties`

Regardless, the whole idea that `torch` folder should somehow depend on `BUILD_CAFFE2` options feels wrong. Perhaps this code needs to be moved to `caffe2` or `aten` folder?
Also, it feels semantically wrong that you want to make any code in `torch` folder depend on `caffe2` build options.

Reply:
You are right, thanks!
I have moved the `BUILD_CAFFE2` build option handling to `caffe2/CMakeLists.txt`. That not only have improved code cohesion but also fixed the issue with less lines of code.

The issue I referred to was self-inflicted while I was trying to handle BUILD_CAFFE2 option from `torch/CMakeLists.txt`. In that attempt, BUILD_CAFFE2 didnt read the correct value defined at the main CMakeLists.txt.